> I am unsure why buildkite fails - neither `scripts/twister --inline-logs -N -v --integration --retry-failed 3 -T tests/kernel/lifo/lifo_usage/` nor `scripts/twister --inline-logs -N -v --integration --retry-failed 3 -T samples/kernel/condition_variables/condvar/` errors on my own machine.

There seems to be some unrelated issues in the kernel side and some tests are indeed failing. Dunno what is wrong there.
But there is one net_pkt failure here
```
/workdir/zephyr/subsys/net/ip/net_pkt.c: In function 'net_pkt_print_frags':
/workdir/zephyr/subsys/net/ip/net_pkt.c:347:45: error: 'buf' undeclared (first use in this function)
  347 |     count, frag, frag->len, net_buf_max_len(buf),
      |                                             ^~~
```

--- a/include/net/net_pkt.h
+++ b/include/net/net_pkt.h
@@ -1568,6 +1568,9 @@ void net_pkt_append_buffer(struct net_pkt *pkt, struct net_buf *buffer);
 /**
  * @brief Get available buffer space from a pkt
  *
+ * @note Reserved bytes (headroom) in any of the fragments are not considered to
+ *       be available.
+ *
  * @param pkt The net_pkt which buffer availability should be evaluated
  *
  * @return the amount of buffer available
@@ -1576,6 +1579,9 @@ size_t net_pkt_available_buffer(struct net_pkt *pkt);
 
 /**
  * @brief Get available buffer space for payload from a pkt
+ *
+ * @note Reserved bytes (headroom) in any of the fragments are not considered to
+ *       be available.
  *
  * @details Unlike net_pkt_available_buffer(), this will take into account
  *          the headers space.

--- a/subsys/net/ip/net_pkt.c
+++ b/subsys/net/ip/net_pkt.c
@@ -344,7 +344,7 @@ void net_pkt_print_frags(struct net_pkt *pkt)
 		frag_size = frag->size;
 
 		NET_INFO("[%d] frag %p len %d max len %u size %d pool %p",
-			 count, frag, frag->len, net_buf_max_len(buf),
+			 count, frag, frag->len, net_buf_max_len(frag),
 			 frag_size, net_buf_pool_get(frag->pool_id));
 
 		count++;
@@ -1020,13 +1020,13 @@ static size_t pkt_estimate_headers_length(struct net_pkt *pkt,
 	return hdr_len;
 }
 
-static size_t pkt_get_size(struct net_pkt *pkt)
+static size_t pkt_get_max_len(struct net_pkt *pkt)
 {
 	struct net_buf *buf = pkt->buffer;
 	size_t size = 0;
 
 	while (buf) {
-		size += buf->size;
+		size += net_buf_max_len(buf);
 		buf = buf->frags;
 	}
 
@@ -1039,7 +1039,7 @@ size_t net_pkt_available_buffer(struct net_pkt *pkt)
 		return 0;
 	}
 
-	return pkt_get_size(pkt) - net_pkt_get_len(pkt);
+	return pkt_get_max_len(pkt) - net_pkt_get_len(pkt);
 }
 
 size_t net_pkt_available_payload_buffer(struct net_pkt *pkt,
@@ -1118,7 +1118,7 @@ int net_pkt_alloc_buffer(struct net_pkt *pkt,
 	}
 
 	/* Verifying existing buffer and take into account free space there */
-	alloc_len = pkt_get_size(pkt) - net_pkt_get_len(pkt);
+	alloc_len = net_pkt_available_buffer(pkt);
 	if (!alloc_len) {
 		/* In case of no free space, it will account for header
 		 * space estimation
@@ -1495,7 +1495,8 @@ static void pkt_cursor_jump(struct net_pkt *pkt, bool write)
 
 	cursor->buf = cursor->buf->frags;
 	while (cursor->buf) {
-		size_t len = write ? cursor->buf->size : cursor->buf->len;
+		const size_t len =
+			write ? net_buf_max_len(cursor->buf) : cursor->buf->len;
 
 		if (!len) {
 			cursor->buf = cursor->buf->frags;
@@ -1520,7 +1521,7 @@ static void pkt_cursor_advance(struct net_pkt *pkt, bool write)
 		return;
 	}
 
-	len = write ? cursor->buf->size : cursor->buf->len;
+	len = write ? net_buf_max_len(cursor->buf) : cursor->buf->len;
 	if ((cursor->pos - cursor->buf->data) == len) {
 		pkt_cursor_jump(pkt, write);
 	}
@@ -1536,9 +1537,10 @@ static void pkt_cursor_update(struct net_pkt *pkt,
 		write = false;
 	}
 
-	len = write ? cursor->buf->size : cursor->buf->len;
+	len = write ? net_buf_max_len(cursor->buf) : cursor->buf->len;
 	if (length + (cursor->pos - cursor->buf->data) == len &&
-	    !(net_pkt_is_being_overwritten(pkt) && len < cursor->buf->size)) {
+	    !(net_pkt_is_being_overwritten(pkt) &&
+	      len < net_buf_max_len(cursor->buf))) {
 		pkt_cursor_jump(pkt, write);
 	} else {
 		cursor->pos += length;
@@ -1563,7 +1565,8 @@ static int net_pkt_cursor_operate(struct net_pkt *pkt,
 		}
 
 		if (write && !net_pkt_is_being_overwritten(pkt)) {
-			d_len = c_op->buf->size - (c_op->pos - c_op->buf->data);
+			d_len = net_buf_max_len(c_op->buf) -
+				(c_op->pos - c_op->buf->data);
 		} else {
 			d_len = c_op->buf->len - (c_op->pos - c_op->buf->data);
 		}
@@ -1693,7 +1696,7 @@ int net_pkt_copy(struct net_pkt *pkt_dst,
 		}
 
 		s_len = c_src->buf->len - (c_src->pos - c_src->buf->data);
-		d_len = c_dst->buf->size - (c_dst->pos - c_dst->buf->data);
+		d_len = net_buf_max_len(c_dst->buf) - (c_dst->pos - c_dst->buf->data);
 		if (length < s_len && length < d_len) {
 			len = length;
 		} else {

--- a/tests/net/net_pkt/src/main.c
+++ b/tests/net/net_pkt/src/main.c
@@ -541,6 +541,7 @@ struct net_buf b5 = {
 	.data  = b5_data,
 	.len   = 0,
 	.size  = 0,
+	.__buf  = b5_data,
 };
 
 uint8_t b4_data[4] = "mnop";
@@ -550,11 +551,14 @@ struct net_buf b4 = {
 	.data  = b4_data,
 	.len   = sizeof(b4_data) - 2,
 	.size  = sizeof(b4_data),
+	.__buf  = b4_data,
 };
 
 struct net_buf b3 = {
 	.frags = &b4,
 	.ref   = 1,
+	.data  = NULL,
+	.__buf  = NULL,
 };
 
 uint8_t b2_data[8] = "efghijkl";
@@ -564,6 +568,7 @@ struct net_buf b2 = {
 	.data  = b2_data,
 	.len   = 0,
 	.size  = sizeof(b2_data),
+	.__buf  = b2_data,
 };
 
 uint8_t b1_data[4] = "abcd";
@@ -573,6 +578,7 @@ struct net_buf b1 = {
 	.data  = b1_data,
 	.len   = sizeof(b1_data) - 2,
 	.size  = sizeof(b1_data),
+	.__buf  = b1_data,
 };
 
 void test_net_pkt_copy(void)

